### ISSUE TITLE
icon overlays not all visible

### ISSUE THREAD

**Issue by tortino** [tablacus/TablacusExplorer#140]
My environment:
TE64 18.5.21 Win 10.0.16299 WS 100 Admin IE 11 it_it 96 McAfee VirusScan Enterprise Windows Defender
foldersettings,multiprocess,mouse,key,mainmenu,titlebar,tabs,tabplus,statusbar,innerback,innerforward,inneraddressbar,innersearchbar,innerup,innerrefresh,innernewfolder,gridlines,split,undoclosetab,touchex,findfiles,delete,fastlistmode,autoupdate,addonsupdater,flat,filterbutton,favorites,usercss.

On my PC I have installed different programs that use overlay icons, like OneDrive, DropBox, and TortoiseSVN. I know that Windows File Explorer has a limitation on the number of overlay icons that can show (see registry key HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\ShellIconOverlayIdentifiers).
On my PC I cannot see some overlays, using both Windows File Explorer and Tablacus.
I have discovered that running Tablacus as administrator works, and I can see all overlay icons (don't know why). However I cannot run it as admin because of the separate issue on drag and drop.

I have also found that other programs can show all the icons, like this one https://www.zabkat.com/x2-64bit.htm (shareware, with a trial). I can see all icon even without running it ad admin.

So is it possibile to modify Tablacus to add support for all icons? (maybe a new addon?)

Thanks.

**Comment by tablacus** [comment]
I haven't encountered this.

My environment:
TE64 18.5.21 Win 10.0.17134 WS 300 IE 11 en_us 96 Windows Defender
mouse,key,addressbar,filterbar,back,forward,up,treeview,toolbar,mainmenu,titlebar,tabplus,multithread,remember,undoclosetab,badge,selectnonexistent,nonexistentoverlay,quickmenu,addonsupdater,usercss

![image](https://user-images.githubusercontent.com/5156977/40361366-0ec47a3e-5e04-11e8-861d-1cafe45d1a29.png)

Thank you,

**Comment by tortino** [comment]
To reproduce the bug you must install more windows applications that use overlay icons, like google drive, one drive, tortoisesvn, drop box... when the total number of overlays is more that 15 (if I remembert correctly...) you see that the last installed application doesn't show overlays (because the application installed first are using all overlay "slots". 
From what I have seen some application (I think one drive and maybe TortoiseSVN, and possibly others) rename the registry keys adding some spaces character at the beginning of each key, so that they are loaded before other applications...
This is my registry overlay key (you see the spaces at the beginning of  keys):
![image](https://user-images.githubusercontent.com/26486758/40376949-f5c3e592-5def-11e8-9495-e9a8dafc480f.png)
To add more detail this my tablacus not showing all overlays (missing one on branches folder):
![image](https://user-images.githubusercontent.com/26486758/40377227-b077f09a-5df0-11e8-8e6d-1d7aa750c0aa.png)
When I run tablacus ad administrator I can see ALL overlays:
![image](https://user-images.githubusercontent.com/26486758/40377285-d195cdd8-5df0-11e8-8151-55fa274c9797.png)

**Comment by tablacus** [comment]
Can you see branches overlay icon in Explorer?
I think overlay icon limitation is 11.
Try moving OneDrive backwards.

Best regards,

**Comment by tortino** [comment]
Hello,
no I cannot see in Windows Explorer, but I can see it in other commercial file managers, like xplorer2  or directory opus (I'm just using trial version to see how they work).
In any case moving OneDrive backward is not a real solution, because then I won't see it's icons.
I was asking if it was possible to create an addon for tablacus, to do whatever the commercial file managers do. I think that maybe they read the registry, and show the overlays in a different manner from Windows, so they can show all overlays.
Thanks.

**Comment by tablacus** [comment]
I've made an add-on "Icon overlay".
Please try to install it.

Best regards,

**Comment by tortino** [comment]
Hi, I have tried the addon, but doesn't work: the overlay on "branches" doesn't show up.
Also what is the "base" option? Should I change it?
Is there something I can do to help you fix the addon?
Maybe you can create a special version of the addon with many additional logs, to see what is happening on my system?

**Comment by tablacus** [comment]
Things don't turn out the way I expected.
Please try setting "Base" to 0.
"Base" is the number to skip the overlay icon displayed by default.

![image](https://user-images.githubusercontent.com/5156977/40516754-181d58ea-5fed-11e8-9cfe-b917dfff7930.png)

Best regards,

**Comment by tortino** [comment]
yes, setting base=0 works, now I can see the overlay on "branches" folder.
I'll keep testing the addon and report any issue, but for now it seems to work correctly.
thanks!

**Comment by tablacus** [comment]
Perhaps the icon cache is broken.
Please try to clear the icon cache just in case.

Command-line: Clear Icon cache.
`ie4uinit.exe -ClearIconCache`

Best regards,

**Comment by tortino** [comment]
no changes after ClearIconCache.
however it seems to be working correctly with base=0

**Comment by tablacus** [comment]
I see.
That will do for the time being.

Thank you,

**Comment by rossHell** [comment]
hello, 
I have the same issue actually with overlay icons of TortoiseSVN. Icons for OneDrive are visible but not on the file icons, but under the column "Status" 

I have installed "icon overlay" add-on but it makes Tablacus crashing. Is there any reason for that ? any add-on which can conflict with this one ? Icons are visible from Windows Explorer.
